File: kanbanBoard-Front End/src/app/classes/task.ts
export class Task {
    taskId!:number
    taskName!:string
    taskDescription!:string
    taskImage!:string
    // is the task completed or not
    taskCompletionStatus!:boolean
    // status of task from list of available task statuses
    taskStatus!:string
    // priority of task from list of available priorities
    taskPriority!:string
    //email id of the member the task is assigned to
    taskAssignedTo!:string
    dueDate!:Date
}

File: kanbanBoard-Back End/boards/src/main/java/com/niit/boards/model/Task.java
package com.niit.boards.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.text.SimpleDateFormat;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    @Id
    private int taskId;
    // Name
    private String taskName;
    // A description on what to do
    private String taskDescription;
    // task image url, if any
    private String taskImage;
    // true if completed, else false
    private boolean taskCompletionStatus;
    // can be on track,in progress etc
    private String taskStatus;
    // can be high, low, etc.
    private String taskPriority;
    //give user emailId
    private String taskAssignedTo;
    //The deadline date for task completion
    private Date dueDate;

}
